src/Http/Http.Extensions/src/HttpRequestJsonExtensions.cs

```diff
@@ -76,6 +81,47 @@ public static class HttpRequestJsonExtensions
         }
     }
 
+    /// <summary>
+    /// Read JSON from the request and deserialize to the specified type.
+    /// If the request's content-type is not a known JSON type then an error will be thrown.
+    /// </summary>
+    /// <param name="request">The request to read from.</param>
+    /// <param name="jsonTypeInfo">Metadata about the type to convert.</param>
+    /// <param name="cancellationToken">A <see cref="CancellationToken"/> used to cancel the operation.</param>
+    /// <returns>The deserialized value.</returns>
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
+    public static async ValueTask<TValue?> ReadFromJsonAsync<TValue>(
```

Comment:
I believe runtime has JSON extension methods on HttpClient. What have they done here? Is this change mirrored from them or have they stuck with one overload?

Reply:
This mirrors HttpClient overload changes: https://github.com/dotnet/runtime/blob/c1d6d849820957918c69f2d1c1801cf3cb2578c9/src/libraries/System.Net.Http.Json/ref/System.Net.Http.Json.cs